odxtools/diagservice.py

```diff
@@ -167,7 +167,10 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
         for cpr in self.comparam_refs:
             cpr._resolve_odxlinks(odxlinks)
 
-        self._request = odxlinks.resolve(self.request_ref)
+        if TYPE_CHECKING:
+            self._request = odxlinks.resolve(self.request_ref, Request)
+        else:
+            self._request = odxlinks.resolve(self.request_ref)
```

Comment:
why?

Reply:
without it, strict mypy produces
```
odxtools/diagservice.py:110: error: Returning Any from function declared to return "Request | None"  [no-any-return]
```

although this does not apply in this specific location, we cannot simply use the type safe branch everywhere because this will often produce cyclic imports. Anyway for this location it can be fixed, thanks for the pointing .